### PR TITLE
chore(config): multiple changes to configuration

### PR DESCRIPTION
**Pull request by GregRos**
Config files
- Changed format of files from CommonJS to JSON
- Added vscode config files
- Added recommended extensions
- Moved husky configuration to package.json

Prettier
- Added plugins for jsdoc and packagejson

Eslint
- Removed @gregros/eslint-config and @typescript-eslint/recommended
- Added rules from the above manually

Package.json
- Applied prettier plugin
- Added tags
- Changed description to explain what the library does
- Added engines property
- Added contributors

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    "ecmaVersion": 2022,
+    "sourceType": "commonjs"
+  },
+  "rules": {
+    //! UNSORTED RULES
+    // Place any additional rules either here at the top, or in the appropriate section below.
+    "@typescript-eslint/consistent-type-imports": "error",
+    "@typescript-eslint/consistent-type-exports": "error",
+
+    //! Disable formatting rules since we're using prettier.
```
Comment (mikavilpas) [inline .eslintrc.json:19]: suggestion(optional): one option to sidestep prettier+eslint issues can also be to use https://prettier.io/docs/en/integrating-with-linters.html and specifically `eslint-config-prettier`. I think this is fine though, and I don't have a strong optinion on this 🙂

File: README.md [hunk 2]
```diff
@@ -112,7 +112,21 @@ if (result.isOkay) {
 
 Dealing with failure is hard, whether you’re realizing you can never be a father, or whether you’re a parser combinator library. You need to be able to signal syntax errors to the user, but also be capable of parsing alternative expressions.
 
-`parjs` handles it by using the SHF or 😕😬💀 system, which I just made up. It recognizes three kinds of failures:
```
Comment (mikavilpas) [inline README.md:115]: 😄 I loved this joke, but maybe it's not that funny for everyone and it's good to change it now.

File: README.md [hunk 3]
```diff
@@ -112,7 +112,21 @@ if (result.isOkay) {
...
+
+#### **⛔Nope**
+
+A **⛔Nope** means that the parser looked at the input and decided it won’t be able to handle it.
+
+Rejections aren’t logged and don’t collect any diagnostic information when the failure happens.
+
+#### **😬Panic**
+
+This usually indicates a syntax error. An **😬Panic**
```
Comment (mikavilpas) [inline README.md:127]: typo: I think this should read "A panic"

**Review by mikavilpas — APPROVED** [review]
🚀

File: README.md (hunk 3 above)
Comment (GregRos) [reply README.md:127]: oh crap that should never have even entered the PR... I didn't mean to do readme changes

File: README.md (hunk 2 above)
Comment (GregRos) [reply README.md:115]: yeah you're right

**Pull request merged**